src/components/ExpandCollapse/ExpandCollapse.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useRef, forwardRef } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import "./ExpandCollapse.scss";
+import Icon from "../Icon/Icon";
+import DropdownChevronDown from "../Icon/Icons/components/DropdownChevronDown";
+
+const ExpandCollapse = forwardRef(
+  ({ children, headerComponentRenderer, className, defaultOpenState, iconSize }, ref) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+
+    const [isOpen, setIsOpen] = useState(defaultOpenState);
+
+    const toogleExpand = () => {
+      setIsOpen(!isOpen);
+    };
+
+    return (
+      <div ref={mergedRef} className={cx("expand-collapse--wrapper", className)}>
+        <div className="expand-collapse">
+          <div className="expand-collapse__header expand-collapse__section" onClick={toogleExpand}>
+            {headerComponentRenderer && headerComponentRenderer()}
```

Comment:
this should be onClickCapture - as the inner component might catch the event before it bubbels